### DISCUSSION TITLE
Cannot upload images

### DISCUSSION THREAD

**Discussion by oblomow** [LycheeOrg/Lychee#2328] [Bugs] [answered]
### Lychee version

v5.1.2

### Which PHP version are you using?

PHP 8.2

### Detailed description of the problem

uploading images through the webinterface does not work. the uploading pop-up does not end/dissappear

### Steps to reproduce the issue

1. logon to webinterface
2. click upload
3. watch pop-up 'uploading'
4. see that uploading does not finish

### Diagnostics [REQUIRED]

Diagnostics
-----------
Info: Latest version of PHP is 8.3
Info: Found 52 small that could be generated.
     You can use `php artisan lychee:generate_thumbs small 52` to generate them.
Info: Found 1190 small2x that could be generated.
     You can use `php artisan lychee:generate_thumbs small2x 1190` to generate them.
Info: Found 105 medium that could be generated.
     You can use `php artisan lychee:generate_thumbs medium 105` to generate them.
Info: Found 159 medium2x that could be generated.
     You can use `php artisan lychee:generate_thumbs medium2x 159` to generate them.
Info: Found 29551 small images without filesizes.
     You can use `php artisan lychee:variant_filesize 29551` to compute them.

System Information
------------------
Lychee Version (git):                    master (5435ba4) -- Up to date (1 days ago).
DB Version:                              5.1.2

composer install:                        --no-dev
APP_ENV:                                 local
APP_DEBUG:                               false
APP_URL:                                 set
APP_DIR:                                 default
LOG_VIEWER_ENABLED:                      true
LIVEWIRE_ENABLED:                        true

System:                                  FreeBSD
PHP Version:                             8.2.13
PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                Europe/Amsterdam
Max uploaded file size:                  100M
Max post size:                           100M
Livewire chunk size:                     12.00 MB
Max execution time:                      0
MySQL Version:                           8.0.35

exec() Available:                        yes
Imagick Available:                       1
Imagick Enabled:                         1
Imagick Version:                         1808
GD Version:                              2.3.3
Number of foreign key:                   14 found.

Config Information
------------------
version:                                 050102
check_for_updates:                       1
sorting_photos_col:                      taken_at
sorting_photos_order:                    ASC
sorting_albums_col:                      title
sorting_albums_order:                    DESC
imagick:                                 1
skip_duplicates:                         0
small_max_width:                         0
small_max_height:                        360
medium_max_width:                        1920
medium_max_height:                       1080
lang:                                    en
image_overlay_type:                      date
default_license:                         CC-BY-NC-ND-4.0
compression_quality:                     90
grants_full_photo_access:                1
delete_imported:                         0
mod_frame_enabled:                       1
mod_frame_refresh:                       30
thumb_2x:                                1
small_2x:                                1
medium_2x:                               1
landing_page_enable:                     1
site_owner:                              
landing_title:                           
landing_subtitle:                        
sm_facebook_url:                         
sm_flickr_url:                           
sm_twitter_url:                          
sm_instagram_url:                        
sm_youtube_url:                          
landing_background:                      dist/finland.jpg
site_title:                              
footer_show_copyright:                   1
site_copyright_begin:                    1984
site_copyright_end:                      2024
footer_additional_text:                  
footer_show_social_media:                0
search_public:                           0
SL_enable:                               0
SL_for_admin:                            0
recent_age:                              1
grants_download:                         1
photos_wraparound:                       1
map_display:                             0
zip64:                                   1
map_display_public:                      0
map_provider:                            OpenStreetMap.org
force_32bit_ids:                         0
map_include_subalbums:                   0
update_check_every_days:                 3
has_exiftool:                            1
share_button_visible:                    1
import_via_symlink:                      0
has_ffmpeg:                              0
location_decoding:                       0
location_decoding_timeout:               30
location_show:                           0
location_show_public:                    0
rss_enable:                              0
rss_recent_days:                         7
rss_max_items:                           100
prefer_available_xmp_metadata:           0
editor_enabled:                          1
lossless_optimization:                   0
swipe_tolerance_x:                       150
swipe_tolerance_y:                       250
local_takestamp_video_formats:           .avi|.mov
log_max_num_line:                        1000
unlock_password_photos_with_url_param:   0
nsfw_visible:                            1
nsfw_blur:                               0
nsfw_warning:                            0
nsfw_warning_admin:                      0
nsfw_banner_override:                    
map_display_direction:                   1
album_subtitle_type:                     oldstyle
upload_processing_limit:                 4
new_photos_notification:                 1
legacy_id_redirection:                   1
zip_deflate_level:                       6
SA_enabled:                              1
default_album_protection:                1
allow_username_change:                   1
album_decoration:                        layers
album_decoration_orientation:            row
auto_fix_orientation:                    1
use_job_queues:                          0
random_album_id:                         starred
use_last_modified_date_when_no_exif_date: 0
ffmpeg_path:                             
ffprobe_path:                            
layout:                                  justified
date_format_photo_thumb:                 M j, Y, g:i:s A e
date_format_photo_overlay:               M j, Y, g:i:s A e
date_format_sidebar_uploaded:            M j, Y, g:i:s A e
date_format_sidebar_taken_at:            M j, Y, g:i:s A e
date_format_hero_min_max:                F Y
date_format_hero_created_at:             M j, Y, g:i:s A T
date_format_album_thumb:                 M Y
upload_chunk_size:                       0
nsfw_banner_blur_backdrop:               0
search_pagination_limit:                 1000
search_minimum_length_required:          4
photo_layout_justified_row_height:       320
photo_layout_masonry_column_width:       300
photo_layout_grid_column_width:          250
photo_layout_square_column_width:        200
photo_layout_gap:                        12
display_thumb_album_overlay:             always
display_thumb_photo_overlay:             hover
use_album_compact_header:                0
default_album_thumb_aspect_ratio:        1/1
login_button_position:                   left
back_button_enabled:                     0
back_button_text:                        Return to Home
back_button_url:                         /
current_job_processing_visible:          1

### Browser & System [REQUIRED]

firefox 124.0.1

server freebsd 14.0

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
Thank you for the report, I will try to investigate a bit later to see if I can reproduce it. In theory, the window is supposed to close if no uploads failed.

**Reply by oblomow** [reply]
thank you. The window just stays there for at leat 15 minutes. ( still counting). Let me know if I can do any additional testing to help you.

**Comment by oblomow** [comment]
Since my lychee installation is currently broken and I want to move forward: I made a snapshot of the installation and did an manual update (https://lycheeorg.github.io/docs/update.html) to see if that fixes things. However this results in a database error. Buit this might be easier to solve/diagnose:

```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p.is_public' in 'where clause'
select
	`albums`.*,
	(
		select
			MIN(taken_at)
		from
			`albums` as `a`
			inner join `photos` on `album_id` = `a`.`id`
		where
			`a`.`_lft` >= `albums`.`_lft`
			and `a`.`_rgt` <= `albums`.`_rgt`
			and `taken_at` is not null
	) as `min_taken_at`,
	(
		select
			MAX(taken_at)
		from
			`albums` as `a`
			inner join `photos` on `album_id` = `a`.`id`
		where
			`a`.`_lft` >= `albums`.`_lft`
			and `a`.`_rgt` <= `albums`.`_rgt`
			and `taken_at` is not null
	) as `max_taken_at`,
	(
		select
			COUNT(*)
		from
			`albums` as `a`
			left join (
				select
					`base_album_id`,
					`is_link_required`,
					`password`
				from
					`access_permissions` as `computed_access_permissions`
				where
					`user_id` is null
			) as `computed_access_permissions` on `computed_access_permissions`.`base_album_id` = `a`.`id`
		where
			`a`.`parent_id` = `albums`.`id`
			and (
				`computed_access_permissions`.`is_link_required` = 0
			)
	) as `num_children`,
	(
		select
			COUNT(*)
		from
			`photos` as `p`
			left join (
				select
					`base_album_id`,
					`is_link_required`,
					`password`
				from
					`access_permissions` as `computed_access_permissions`
				where
					`user_id` is null
			) as `computed_access_permissions` on `computed_access_permissions`.`base_album_id` = `p`.`album_id`
		where
			`p`.`album_id` = `albums`.`id`
			and (
				`computed_access_permissions`.`is_link_required` = 0
				or `p`.`is_public` = 1
			)
	) as `num_photos`
from
	`albums`
	left join (
		select
			`base_albums`.`id`,
			`base_albums`.`owner_id`,
			`base_albums`.`title`,
			`base_albums`.`created_at`,
			`base_albums`.`description`
		from
			`base_albums` as `base_albums`
	) as `base_albums` on `base_albums`.`id` = `albums`.`id`
	left join (
		select
			`base_album_id`,
			`is_link_required`,
			`password`
		from
			`access_permissions` as `computed_access_permissions`
		where
			`user_id` is null
	) as `computed_access_permissions` on `computed_access_permissions`.`base_album_id` = `albums`.`id`
where
	`parent_id` is null
	and (
		`computed_access_permissions`.`is_link_required` = 0
	)
```

**Reply by ildyria** [reply]
Which version of the database are you running ? How do you get this error ?
Which file version are you running ?

If you did a manual update by downloading the zip file of the release, this will break because migrations have been run since (if you were on lastest/alpha/master) where you have difference in code compared to the latest releases. As a result the SQL requests will no longer be matching the state of the database.

**Reply by ildyria** [reply]
Note that if the upload pop up does not close, it is fine as long as all the files are at 100% or marked as finished.

**Comment by oblomow** [comment]
Database? mysql  Ver 8.0.35 for FreeBSD14.0 on amd64 (Source distribution)

What do you mean by 'file version'?

Yes, I did a manual update while being on the master branch. But that was just a quick test. I made a filesystem snapshot, so I can revert easily

I reverted back to the state of the Lychee installation with the pop-up problem. The pop-up stays, but the photo is not uploaded. The album that I'm using stays empty. ('nothing to see here')

**Reply by ildyria** [reply]
but if you reload the page, nothing shows?
Could you check that your web user has write access to the storage folders ?

**Reply by oblomow** [reply]
correct. nothing shows after a reload.

permissions haven't changed

`total 130
drwxr-xr-x  12 www www    19 Jan 26 03:05 .
drwxrwxr-x  21 www www    53 Mar 26 21:59 ..
-rw-r--r--   1 www www  1904 Dec 26 11:36 .htaccess
-rw-r--r--   1 www www   387 Dec 26 11:36 .user.ini
drwxrwxr-x   3 www www     4 Dec 26  2022 .well-known
drwxr-xr-x   2 www www     2 May  3  2020 Lychee-front
drwxr-xr-x   3 www www     4 Jan 15 21:53 build
drwxrwxr-x   3 www www    12 Mar  4 09:31 dist
-rw-r--r--   1 www www 33412 Dec 26 11:36 favicon.ico
drwxr-xr-x   2 www www     5 Dec 26 11:36 fonts
drwxr-xr-x   2 www www    19 Dec 26 11:36 img
-rw-r--r--   1 www www  2614 Dec 26 11:36 index.php
drwxr-xr-x   3 www www     3 Apr 20  2020 installer
-rw-r--r--   1 www www   254 Dec 26 11:36 robots.txt
lrwxr-xr-x   1 www www    21 Dec 26 11:36 storage -> ../storage/app/public
drwxrwsr-x   2 www www     3 Dec 26 11:36 sym
drwxrwsr-x  13 www www    13 Dec 26 19:27 uploads
drwxrwxr-x   3 www www     3 May 19  2023 vendor
-rw-r--r--   1 www www  1193 Dec 26 11:36 web.config

`

**Comment by oblomow** [comment]
> If you did a manual update by downloading the zip file of the release, this will break because migrations have been run since (if you were on lastest/alpha/master) where you have difference in code compared to the latest releases. As a result the SQL requests will no longer be matching the state of the database.

So if I wait until stable catches up with the master branch, I can try this again? Do you know when this will happen?

**Reply by ildyria** [reply]
Sorry for the late reply, just came back from a week of conference. :(
stable should catch up to master... As soon as some people would be keen enough to review the pull requests. :(
I made a cry for help on reddit, we are pretty open as a team, but it kinda feels like this.

![image](https://imgs.xkcd.com/comics/dependency_2x.png)

In the end we may just need to drop the peer review process. :(

**Comment by oblomow** [comment]
New test: uploading through 'import from a link' does work. (edit: even multiple files at a time)

**Reply by ildyria** [reply]
that is very weird. :(

**Reply by oblomow** [reply]
Yes. I'll wait until stable is available and try again. I don't want to waste your time hunting down a very obscure bug on my system.

**Comment by oblomow** [comment, marked as answer]
succces! downloading stable 5.2.0 solved the problem. I will stay away from the git master branch ( and compiling it myself). Thanks!
